.github/workflows/CI.yml

```diff
@@ -36,63 +43,75 @@ jobs:
     - name: Build
       working-directory: build
       run: cmake --build . -j $env:NUMBER_OF_PROCESSORS --config ${{ matrix.configuration }}
-    - uses: actions/upload-artifact@v1
+    - uses: actions/upload-artifact@v2
       if: matrix.toolset == 'v141_xp'
       with:
         name: openag-${{ runner.os }}-${{ matrix.configuration }}
         path: build\${{ matrix.configuration }}\client.dll
+        if-no-files-found: error
 
   build-linux-gcc:
-    runs-on: ubuntu-18.04
-    env:
-      CC: gcc-${{ matrix.gcc-ver }}
-      CXX: g++-${{ matrix.gcc-ver }}
+    name: build-linux (gcc-${{ matrix.gcc-ver }}, ${{ matrix.configuration }})
+    runs-on: ubuntu-latest
```

Comment:
This needs to be `ubuntu-18.04` for two reasons. First, `-latest` automatically breaks CI when it changes, I'm not a big fan of that. Second, the built `client.so` links to symbols from the distribution glibc, `ubuntu-20.04` has a newer glibc, so compiling on `ubuntu-20.04` makes `cilent.so` incompatible with Ubuntu 18.04 and other slightly outdated distributions.